Docs: Changed OpenVINO Runtime User Guide integration (#10187)
openvinotoolkit/openvino: docs/OV_Runtime_UG/model_representation.md

For details on how to build a model in OpenVINO™ Runtime, see the [Build a Model in OpenVINO™ Runtime](@ref build_model) section.

> OpenVINO™ Runtime allows to use tensor names or indexes to work wit model inputs/outpus. To get model input/output ports you can use `ov::Model::inputs()` or `ov::Model::outputs()` respectively.

Reviewer: ```suggestion
OpenVINO™ Runtime allows using tensor names or indexes to work wit model inputs/outputs. To get model input/output ports, use the `ov::Model::inputs()` or `ov::Model::outputs()` methods respectively.
```

> OpenVINO™ Runtime allows using tensor names or indexes to work wit model inputs/outputs. To get model input/output ports, use the `ov::Model::inputs()` or `ov::Model::outputs()` methods respectively.

Author: We have different pictures in MO documentation. @andrew-zaytsev I think we need to add such picture separately.

> OpenVINO™ Runtime allows using tensor names or indexes to work wit model inputs/outputs. To get model input/output ports, use the `ov::Model::inputs()` or `ov::Model::outputs()` methods respectively.

Reviewer 2: let's create a ticket.

> OpenVINO™ Runtime allows using tensor names or indexes to work wit model inputs/outputs. To get model input/output ports, use the `ov::Model::inputs()` or `ov::Model::outputs()` methods respectively.

Author: Created a tiket 80745

@sphinxdirective

.. tab:: C++

    .. doxygensnippet:: docs/snippets/ov_model_snippets.cpp
       :language: cpp
       :fragment: [all_inputs_ouputs]

.. tab:: Python

    .. doxygensnippet:: docs/snippets/ov_model_snippets.py
       :language: python
       :fragment: [all_inputs_ouputs]

@endsphinxdirective

OpenVINO™ Runtime model representation uses special classes to work with model data types and shapes. For data types the `ov::element::Type` is used.

> ## Shapes representation

Reviewer: ```suggestion
## Shapes Representation
```

OpenVINO™ Runtime provides two types for shape representation: 

* `ov::Shape` - Represents static (fully defined) shapes.

* `ov::PartialShape` - Represents dynamic shapes. That means that the rank or some of dimensions are dynamic (undefined). `ov::PartialShape` can be converted to `ov::Shape` using the `get_shape()` method if all dimensions are static; otherwise the conversion raises an exception.
  `ov::PartialShape` can be converted to `ov::Shape` using the `get_shape()` method if all dimensions are static; otherwise, conversion raises an exception.

  @sphinxdirective

  .. tab:: C++

      .. doxygensnippet:: docs/snippets/ov_model_snippets.cpp
         :language: cpp
         :fragment: [ov:partial_shape]

  .. tab:: Python

      .. doxygensnippet:: docs/snippets/ov_model_snippets.py
         :language: python
         :fragment: [ov:partial_shape]

  @endsphinxdirective

>   But in most cases before getting static shape using `get_shape()` method, you need to check that shape is static.

Author: Yes we have this example in the code snippet

## Operations

> The `ov::Op` class represents any abstract operation in the model representation. Use this class to create [custom operations](../Extensibility_UG/add_openvino_ops).

Author: `ov::Node` is a base class of `ov::Op`

To build an `ov::Model` instance from `opset8` operations, include the following files:

> 

Reviewer 2: the text above says about operation sets, but we already have dedicated section.

@sphinxdirective

.. tab:: C++

    .. doxygensnippet:: docs/snippets/ov_model_snippets.cpp
       :language: cpp
       :fragment: [ov:include]

.. tab:: Python

    .. doxygensnippet:: docs/snippets/ov_model_snippets.py
       :language: python
       :fragment: [import]

@endsphinxdirective

The following code demonstrates how to create a simple model:

@sphinxdirective

.. tab:: C++

    .. doxygensnippet:: docs/snippets/ov_model_snippets.cpp
       :language: cpp
       :fragment: [ov:create_simple_model]

.. tab:: Python

    .. doxygensnippet:: docs/snippets/ov_model_snippets.py
       :language: python
       :fragment: [ov:create_simple_model]

@endsphinxdirective

The following code creates a model with several outputs:

@sphinxdirective

.. tab:: C++

    .. doxygensnippet:: docs/snippets/ov_model_snippets.cpp
       :language: cpp
       :fragment: [ov:create_advanced_model]

.. tab:: Python

    .. doxygensnippet:: docs/snippets/ov_model_snippets.py
       :language: python
       :fragment: [ov:create_advanced_model]

@endsphinxdirective

## Model debug capabilities

OpenVINO™ provides several debug capabilities:

>    - To receive additional messages about applied graph modifications, rebuild the OpenVINO™ Runtime library with the `-DENABLE_OPENVINO_DEBUG=ON` option.

Reviewer 2: should we replace `graph` with `model` everywhere?

   - Model can be visualized to image from the xDot format:

    @sphinxdirective

    .. tab:: C++

        .. doxygensnippet:: docs/snippets/ov_model_snippets.cpp
           :language: cpp
           :fragment: [ov:visualize]

    .. tab:: Python

        .. doxygensnippet:: docs/snippets/ov_model_snippets.py
           :language: python
           :fragment: [ov:visualize]

    @endsphinxdirective

    `ov::pass::VisualizeTree` can be parametrized via environment variables:

    ```
    OV_VISUALIZE_TREE_OUTPUT_SHAPES=1       - visualize shapes
    OV_VISUALIZE_TREE_OUTPUT_TYPES=1        - visualize types
    OV_VISUALIZE_TREE_MIN_MAX_DENORMAL=1    - pretty denormal values
    OV_VISUALIZE_TREE_RUNTIME_INFO=1        - print runtime information
    OV_VISUALIZE_TREE_IO=1                  - print I/O ports
    OV_VISUALIZE_TREE_MEMBERS_NAME=1        - print member names

>     ```

Reviewer 2: this is not rendered well.
